Assets/Plugins/Source/Editor/Windows/CreatePackageWindow.cs

```diff
@@ -262,15 +290,29 @@ private async void StartCreatePackageAsync(UPMUtility.PackageType type)
             _createPackageCancellationTokenSource = new();
             _operationInProgress = true;
 
+            _progressUpdateThread = new Thread(() => UpdateProgressBar(_createPackageCancellationTokenSource.Token));
+            _progressUpdateThread.Start();
+
             var progressHandler = new Progress<UnityPackageCreationUtility.CreatePackageProgressInfo>(value =>
             {
                 var fileCountStrSize = value.TotalFilesToCopy.ToString().Length;
                 string filesCopiedStrFormat = "{0," + fileCountStrSize + "}";
                 var filesCopiedCountStr = String.Format(filesCopiedStrFormat, value.FilesCopied);
                 var filesToCopyCountStr = String.Format(filesCopiedStrFormat, value.TotalFilesToCopy);
 
-                _progress = value.SizeOfFilesCopied / (float)value.TotalSizeOfFilesToCopy;
-                _progressText = $"{filesCopiedCountStr} out of {filesToCopyCountStr} files copied";
+                lock (_progressLock)
+                {
+                    float newActualProgress = value.SizeOfFilesCopied / (float)value.TotalSizeOfFilesToCopy;
+
+                    // Just to guarantee that the progress is increasing
+                    if (newActualProgress > _actualProgress)
+                    {
+                        _actualProgress = newActualProgress;
+                    }
+
+                    _progressText = $"{filesCopiedCountStr} out of {filesToCopyCountStr} files copied";
```

Comment:
Since progress is measured in terms of size rather than files remaining, would it make sense to also include information about the number of files remaining? It surprised me when the progress bar was half-way done while being one fourth of the files remaining, then it eventually made sense as it took more time.

Reply:
I'm not certain I understand - the progress bar _is _being updated here to tell you how many files are left. I'm going to assume you're calling out the discrepancy between having the progress of the bar be based on file size, while the text indicates how many files. I chose to make this distinction because it provided the best experience upon experimentation, since the alternative would have it stuck on a single file for a while if it's particularly large.